### PR TITLE
Document Share Target API

### LINKED ISSUE #126: OWD Project: Document the Web Share Target API
### Problem statement

https://w3c.github.io/web-share-target/ needs documentation

It’s supported by Chromium on Android and Edge on Windows. 

CC: @robnyman 

### Proposed solutions

See task list.

### Task list

- [ ] Create a new page https://developer.mozilla.org/en-US/docs/Web/Manifest/share_target
- [ ] Possibly update https://github.com/mdn/browser-compat-data/blob/main/html/manifest/share_target.json 
- [ ] Update https://developer.mozilla.org/en-US/docs/Web/API/Web_Share_API to at least reference the share_target manifest member.
- [ ] ... possibly more

### Priority assessment

- **Effort**: Small. One new manifest member.
- **Dependencies**: None.
- **Community enablement**: None.
- **Momentum**:  Filed twice with MDN already (see below). Seems like there is demand.
- **Enabling learners**: n/a
- **Enabling professionals**: n/a
- **Underrepresented topics / Ethical web**:  n/a
- **Operational necessities**: no.
- **Addressing needs of the web industry**: Closes another gap so that web apps behave like native apps.

### More information

Filed with MDN before:
- https://github.com/mdn/content/issues/8887
- https://github.com/mdn/content/issues/10291

### LINKED ISSUE #8887: Content suggestion: Share Target Should be Documented
## What is the new suggestion?
Document the [Web Share Target API](https://w3c.github.io/web-share-target/). 

## Why is it important or useful?
The [Web Share API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Share_API) is one side of a coin. It lets web apps to share links, text, and files with other applications on a user's system. The flip side of this is allowing web apps to received such shared resources from other apps, whether native or web based. 

This API is already [supported by Chromium browsers](https://www.chromestatus.com/features#web%20share%20target).

How many pages are likely to be needed? 
One: the API adds a member to the Web App Manifest

How much time do you think this work should take? A few hours? A week? A month?
An hour our two.

Will the work enable learners or professionals to achieve their goals better?
It makes it easier for devices to share resources with web-based apps.

Does it address critical needs in the web industry?
From the [explainer](https://github.com/w3c/web-share-target/blob/main/docs/explainer.md): "This is one of the missing pieces of letting web apps behave like native apps. We've recently gotten new web technologies to let websites be installed alongside native apps, run outside of the browser window, theme the system title bars, and receive push notifications. The one thing many web apps (chat clients, social networking apps) need is a way to receive content from other web apps and native apps."

Is the work an operational necessity, i.e. is not having it a security risk?
No

Does the content help make the web more ethical?
I can't say either way.

### LINKED ISSUE #10291: Issue with "Web app manifests": The document doesn't include the share_target node
MDN URL: https://developer.mozilla.org/en-US/docs/Web/Manifest

#### What information was incorrect, unhelpful, or incomplete?
The document doesn't include the share_target node. Some information about it can be found at the following places: 
- https://w3c.github.io/web-share-target/ 
- https://web.dev/web-share-target/
- https://github.com/w3c/web-share-target/blob/main/docs/explainer.md
- 
#### Specific section or headline?
No specific section. Complete information should be added about it

#### What did you expect to see?
The node included in Example manifest, plus complete information about it. Including browser support.

#### Did you test this? If so, how?
Nothing to test, just expect to see it on the page

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/manifest`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/Manifest
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/manifest/index.md
* Last commit: https://github.com/mdn/content/commit/8fd929ee9ecb9dada20533048ddb04d9e4e95e34
* Document last modified: 2021-10-27T16:11:38.000Z

</details>

### PR DESCRIPTION
**Pull request by Elchi3**
### Description

Add a new page to https://developer.mozilla.org/en-US/docs/Web/Manifest to document the Web Share Target API. 

Spec: https://w3c.github.io/web-share-target/level-2/ (examples mostly taken from there).

### Motivation

Fix https://github.com/mdn/content/issues/10291
Fix https://github.com/mdn/content/issues/8887
Fix https://github.com/openwebdocs/project/issues/126

### CODE REVIEW

File: files/en-us/web/manifest/share_target/index.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    }
+  }
+}
+```
+
+To handle shared file data, see the `POST` example above and the [`FileReader`](/en-US/docs/Web/API/FileReader) API to read the files.
+
+## Security & Privacy
+
+Your PWA can only act as a web share target if it has been installed. See also [How to make PWAs installabe](/en-US/docs/Web/Progressive_web_apps/Installable_PWAs).
```
Comment (wbamberg) [inline files/en-us/web/manifest/share_target/index.md:153]:
```suggestion
Your PWA can only act as a web share target if it has been installed. See also [How to make PWAs installable](/en-US/docs/Web/Progressive_web_apps/Installable_PWAs).
```

**Review by wbamberg — APPROVED** [review]
👍 this looks good to me! @tomayac , would you like to review this as well?

File: files/en-us/web/manifest/share_target/index.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+          "name": "photos",
+          "accept": ["image/svg+xml", ".svg"]
+        }
+      ]
+    }
+  }
+}
+```
+
+To handle shared file data, see the `POST` example above and the [`FileReader`](/en-US/docs/Web/API/FileReader) API to read the files.
```
Comment (tomayac) [inline files/en-us/web/manifest/share_target/index.md:149]:
```suggestion
To handle shared file data, see the `POST` example above and the [`FileReader`](/en-US/docs/Web/API/FileReader) API to read the files. In order to get the files from the service worker context over to client contexts, one solution is to temporarily write the files to the [`Cache`](/en-US/docs/Web/API/Cache) or [IndexedDB](/en-US/docs/Web/API/IndexedDB_API) and then notify its clients using [`Client.postMessage()`](/en-US/docs/Web/API/Client/postMessage).
```

**Review by tomayac — COMMENTED** [review]
Thanks so much for writing this article! It is on a really good track. I have made a couple of suggestions. I see a lot of developers get stuck at the part where incoming files need to be dealt with, so I went a bit into detail. 

(Also see my in-flight pattern for `web.dev` regarding this problem: [staging link](https://deploy-preview-7645--web-dev-staging.netlify.app/patterns/files/receive-shared-files/#using-the-web-share-target-api).).

File: files/en-us/web/manifest/share_target/index.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+```json
+{
+  "share_target": {
+    "action": "/shared-content-receiver/",
+    "method": "GET",
+    "params": {
+      "title": "name",
+      "text": "description",
+      "url": "link",
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply files/en-us/web/manifest/share_target/index.md:65]: I think "link" is correct here, it is the name of the query parameter and not an actual link, right? 
(`?link=https://example.com/` will be in the URL and the API will know that `link` is for `url`)

[the remark this answers is not in the document]
Comment (tomayac) [reply files/en-us/web/manifest/share_target/index.md:65]: Oh, yes of course. Sorry for the noise.

**Comment by Elchi3** [comment]
Thanks for the review, @tomayac! 👍

**Pull request merged**